modules/DesktopStreaming/src/main/java/org/gephi/desktop/streaming/StreamingUIController.java

```diff
@@ -0,0 +1,396 @@
+
+            public void select(Workspace workspace) {
+                model = workspace.getLookup().lookup(StreamingModel.class);
+                if (model == null) {
+                    model = new StreamingModel();
+                    workspace.add(model);
+                }
+                refreshModel();
+            }
+
+            public void unselect(Workspace workspace) {
+            }
+
+            public void close(Workspace workspace) {
+                model = workspace.getLookup().lookup(StreamingModel.class);
+                if (model != null) {
+                    if (model.isMasterRunning()) {
+                        stopMaster();
+                    }
+                    model.removeAllConnections();
+                    refreshModel();
+                }
+            }
+
+            public void disable() {
+                model = null;
+            }
+        });
+
+        if (pc.getCurrentWorkspace() != null) {
+            model = pc.getCurrentWorkspace().getLookup().lookup(StreamingModel.class);
+            if (model == null) {
+                model = new StreamingModel();
+                pc.getCurrentWorkspace().add(model);
+            }
+        }
+    }
+
+    public void setTopComponent(StreamingTopComponent component) {
+        this.component = component;
+        refreshModel();
+    }
+
+    public void refreshModel() {
+        if (model != null) {
+            component.refreshModel(model);
+        }
+    }
+    
+    public StreamingModel getStreamingModel() {
+        return model;
+    }
+
+    /**
+     * Shows the StreamingClientPanel and connect to a StreamingEndpoint using
+     * the entered info.
+     */
+    public void connectToStream() {
+        StreamingClientPanel clientPanel = new StreamingClientPanel();
+
+        ValidationPanel vp = StreamingClientPanel.createValidationPanel(clientPanel);
+        final DialogDescriptor dd = new DialogDescriptor(vp, "Connect to Stream");
+        vp.addChangeListener(new ChangeListener() {
+
+            public void stateChanged(ChangeEvent e) {
+                dd.setValid(!((ValidationPanel) e.getSource()).isProblem());
+            }
+        });
+
+        Object result = DialogDisplayer.getDefault().notify(dd);
+        if (!result.equals(NotifyDescriptor.OK_OPTION)) {
+            return;
+        }
+        StreamingEndpoint endpoint = clientPanel.getGraphStreamingEndpoint();
+        connectToStream(endpoint);
+    }
+
+    /**
+     * Synchronize with the Master: disconnect from it, clean the graph and
+     * reconnect.
+     *
+     * @param connection
+     */
+    public void synchronize(StreamingConnection connection) {
+
+        // Get active graph instance - Project and Graph API
+        ProjectController projectController = Lookup.getDefault().lookup(ProjectController.class);
+        Project project = projectController.getCurrentProject();
+        if (project==null)
+            projectController.newProject();
```

Comment:
Can you clarify if that's actually possible? I'm not sure in which scenario you would have to create the project. I'm asking because that isn't the right way to create a project when their a GUI (it is if done from toolkit for instance) and one should rather use the `ProjectControllerUI.newProject()` in `desktop-project` module instead.